lib/modules/manager/pep723/extract.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { codeBlock } from 'common-tags';
+import { extractPackageFile } from '.';
+
+describe('modules/manager/pep723/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('should extract dependencies', () => {
+      const res = extractPackageFile(
+        codeBlock`
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "requests==2.32.3",
+#   "rich>=13.8.0",
+# ]
+# ///
+`,
+        'foo.py',
+      );
+
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: '==2.32.3',
+            currentVersion: '2.32.3',
+            datasource: 'pypi',
+            depName: 'requests',
+            depType: 'project.dependencies',
+            packageName: 'requests',
+          },
+          {
+            currentValue: '>=13.8.0',
+            datasource: 'pypi',
+            depName: 'rich',
+            depType: 'project.dependencies',
+            packageName: 'rich',
+          },
+        ],
+        extractedConstraints: { python: '>=3.11' },
```

Comment:
Should this also be a dependency so that it can be updated?

Reply:
I'm not entirely sure. [`requires-python`'s purpose](https://packaging.python.org/en/latest/specifications/inline-script-metadata/#script-type) is the same as when used in `pyproject.toml`, it defines the minimum version for which a script is compatible. I believe in some situations (e.g. when building a web application), it would be convenient, but in some others (e.g. building a library that supports multiple Python versions), it might be unwanted.

Currently, PEP 621 manager does not consider `requires-python` as a dependency either (it is just used as a constraint). Regardless of what we chose, we should probably have a consistent implement between the 2.

Reply:
Ok let's leave it as is for now